source/common/upstream/upstream_impl.h

```diff
@@ -490,6 +494,37 @@ class ClusterImplBase : public Cluster, protected Logger::Loggable<Logger::Id::u
   uint64_t pending_initialize_health_checks_{};
 };
 
+/**
+ * Manage priority state of a cluster.
```

Comment:
Is this really used at runtime once state is loaded, or just as a temporary holding ground for priority/host/locality information prior to an update (via `updateClusterPrioritySet`)?